File: app/src/main/java/com/dataqin/testnew/widget/scale/ScaleAdapter.java
package com.dataqin.testnew.widget.scale;

import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;

import java.util.List;

/**
 * Created by wangyanbin
 * 伸缩图片适配器
 */
public class ScaleAdapter extends PagerAdapter {
    private final List<ScaleImageView> data;

    public ScaleAdapter(List<ScaleImageView> data) {
        this.data = data;
    }

    public int getCount() {
        return data.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return (view == object);
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        container.removeView(data.get(position));
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        ScaleImageView img = data.get(position);
        container.addView(img, LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT);
        return img;
    }

}

File: app/src/main/java/com/dataqin/testnew/widget/scale/ScaleAdapter.kt
package com.dataqin.testnew.widget.scale

import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import androidx.viewpager.widget.PagerAdapter
import com.bumptech.glide.Glide
import com.dataqin.common.imageloader.ImageLoader
import com.dataqin.testnew.R

/**
 * Created by wangyanbin
 * 伸缩图片适配器
 */
class ScaleAdapter(var data: List<ScaleImageView>, var fileList: ArrayList<*>) : PagerAdapter() {

    override fun getCount(): Int {
        return data.size
    }

    override fun isViewFromObject(view: View, any: Any): Boolean {
        return view === any
    }

    override fun destroyItem(container: ViewGroup, position: Int, any: Any) {
        container.removeView(data[position])
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val img = data[position]
        ImageLoader.instance.displayImage(img, fileList[position] as String, R.drawable.shape_scale_loading, R.drawable.shape_image_loading)
        container.addView(img, LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT)
        return img
    }

}